app/assets/javascripts/calendars/guider-slot-picker.es6

```diff
@@ -20,13 +20,72 @@
 
       super.start(el);
 
-      this.addEvents();
-      this.generateJSON();
-      this.setupEvents();
+      this.saveWarningMessage = 'You have unsaved changes - Save, or undo the changes.';
+      this.hiddenDataInput = $(this.$el.data('events'));
+      this.saveButton = $('.js-button-save');
+
+      this.addExistingEventsToCalendar();
```

Comment:
It's not strictly the constructor anymore (it's the start method), but I get your jist 👍 - will split out.